### DISCUSSION TITLE
Problem Creating a Website in Azure

### DISCUSSION THREAD

**Discussion by ranjulbn20** [github/india#355] [General]
I am facing a problem in creating a website in Unit 6 of the "Describe core Azure architectural components" module. When trying to create the WordPress website as mentioned in the exercise I'm getting the error shown below.

{
  "code": "InvalidTemplateDeployment",
  "message": "The template deployment failed with multiple errors. Please see details for more information.",
  "details": [
    {
      "code": "RequestDisallowedByPolicy",
      "target": "ASP-learnac9d3d064a3049e79cd75e7b4b-ac7f",
      "message": "Resource 'ASP-learnac9d3d064a3049e79cd75e7b4b-ac7f' was disallowed by policy. Policy identifiers: '[{\"policyAssignment\":{\"name\":\"webapps-assignment\",\"id\":\"/providers/Microsoft.Management/managementGroups/192fe359-e72a-fa7e-dcd5-95e985e7eac9/providers/Microsoft.Authorization/policyAssignments/webapps-assignment\"},\"policyDefinition\":{\"name\":\"app-service-sku-policy\",\"id\":\"/providers/Microsoft.Management/managementGroups/learn-sandbox-prod/providers/Microsoft.Authorization/policyDefinitions/app-service-sku-policy\"},\"policySetDefinition\":{\"name\":\"webapps-initiative\",\"id\":\"/providers/Microsoft.Management/managementGroups/learn-sandbox-prod/providers/Microsoft.Authorization/policySetDefinitions/webapps-initiative\"}}]'.",
      "additionalInfo": [
        {
          "type": "PolicyViolation",
          "info": {
            "policyDefinitionDisplayName": "app-service-sku-policy",
            "policySetDefinitionDisplayName": "webapps-initiative",
            "evaluationDetails": {
              "evaluatedExpressions": [
                {
                  "result": "True",
                  "expressionKind": "Field",
                  "expression": "type",
                  "path": "type",
                  "expressionValue": "Microsoft.Web/serverfarms",
                  "targetValue": "Microsoft.Web/serverFarms",
                  "operator": "Equals"
                },
                {
                  "result": "True",
                  "expressionKind": "Field",
                  "expression": "Microsoft.Web/serverfarms/sku.name",
                  "path": "sku.name",
                  "expressionValue": "P1v2",
                  "targetValue": [
                    "F1",
                    "D1",
                    "Y1",
                    "WS1"
                  ],
                  "operator": "NotIn"
                }
              ]
            },
            "policyDefinitionId": "/providers/Microsoft.Management/managementGroups/learn-sandbox-prod/providers/Microsoft.Authorization/policyDefinitions/app-service-sku-policy",
            "policySetDefinitionId": "/providers/Microsoft.Management/managementGroups/learn-sandbox-prod/providers/Microsoft.Authorization/policySetDefinitions/webapps-initiative",
            "policyDefinitionReferenceId": "app-service-sku-policy_1",
            "policySetDefinitionName": "webapps-initiative",
            "policyDefinitionName": "app-service-sku-policy",
            "policyDefinitionEffect": "Deny",
            "policyAssignmentId": "/providers/Microsoft.Management/managementGroups/192fe359-e72a-fa7e-dcd5-95e985e7eac9/providers/Microsoft.Authorization/policyAssignments/webapps-assignment",
            "policyAssignmentName": "webapps-assignment",
            "policyAssignmentScope": "/providers/Microsoft.Management/managementGroups/192fe359-e72a-fa7e-dcd5-95e985e7eac9"
          }
        }
      ]
    },
    {
      "code": "RequestDisallowedByPolicy",
      "target": "CodingClub-mysqldbserver",
      "message": "Resource 'CodingClub-mysqldbserver' was disallowed by policy. Policy identifiers: '[{\"policyAssignment\":{\"name\":\"webapps-assignment\",\"id\":\"/providers/Microsoft.Management/managementGroups/192fe359-e72a-fa7e-dcd5-95e985e7eac9/providers/Microsoft.Authorization/policyAssignments/webapps-assignment\"},\"policyDefinition\":{\"name\":\"Allowed resource types\",\"id\":\"/providers/Microsoft.Authorization/policyDefinitions/a08ec900-254a-4555-9bf5-e42af04b5c5c\"},\"policySetDefinition\":{\"name\":\"webapps-initiative\",\"id\":\"/providers/Microsoft.Management/managementGroups/learn-sandbox-prod/providers/Microsoft.Authorization/policySetDefinitions/webapps-initiative\"}}]'.",
      "additionalInfo": [
        {
          "type": "PolicyViolation",
          "info": {
            "policyDefinitionDisplayName": "Allowed resource types",
            "policySetDefinitionDisplayName": "webapps-initiative",
            "evaluationDetails": {
              "evaluatedExpressions": [
                {
                  "result": "False",
                  "expressionKind": "Field",
                  "expression": "type",
                  "path": "type",
                  "expressionValue": "Microsoft.DBforMySQL/servers",
                  "targetValue": [
                    "Microsoft.ApiManagement/service",
                    "Microsoft.AlertsManagement/SmartDetectorAlertRules",
                    "Microsoft.Cache/Redis",
                    "Microsoft.Cdn/profiles",
                    "Microsoft.Cdn/profiles/endpoints",
                    "Microsoft.CognitiveServices/accounts",
                    "Microsoft.ContainerRegistry/registries",
                    "Microsoft.ContainerRegistry/registries/webhooks",
                    "Microsoft.DataFactory/factories",
                    "Microsoft.DBforPostgreSQL/servers",
                    "Microsoft.DBforPostgreSQL/serversv2",
                    "Microsoft.DocumentDB/databaseAccounts",
                    "microsoft.insights/actiongroups",
                    "microsoft.insights/components",
                    "Microsoft.KeyVault/vaults",
                    "Microsoft.Logic/workflows",
                    "Microsoft.MixedReality/remoteRenderingAccounts",
                    "Microsoft.Network/connections",
                    "Microsoft.Network/localNetworkGateways",
                    "Microsoft.Network/networkSecurityGroups",
                    "Microsoft.Network/publicIPAddresses",
                    "Microsoft.Network/trafficmanagerprofiles",
                    "Microsoft.Network/virtualNetworkGateways",
                    "Microsoft.Network/virtualNetworks",
                    "Microsoft.Portal/dashboards",
                    "Microsoft.Relay/namespaces",
                    "Microsoft.Resources/resourceGroups",
                    "Microsoft.Search/searchServices",
                    "Microsoft.ServiceBus/namespaces",
                    "Microsoft.ServiceBus/namespaces/authorizationrules",
                    "Microsoft.ServiceBus/namespaces/queues",
                    "Microsoft.ServiceBus/namespaces/queues/authorizationrules",
                    "Microsoft.ServiceBus/namespaces/topics",
                    "Microsoft.ServiceBus/namespaces/topics/authorizationrules",
                    "Microsoft.ServiceBus/namespaces/topics/subscriptions",
                    "Microsoft.ServiceBus/namespaces/topics/subscriptions/rules",
                    "Microsoft.SignalRService/SignalR",
                    "Microsoft.Sql/servers",
                    "Microsoft.Sql/servers/databases",
                    "Microsoft.Storage/storageAccounts",
                    "Microsoft.Web/certificates",
                    "Microsoft.Web/connections",
                    "Microsoft.Web/customApis",
                    "Microsoft.Web/serverFarms",
                    "Microsoft.Web/serverfarms/workers",
                    "Microsoft.Web/sites",
                    "Microsoft.Web/sites/instances",
                    "Microsoft.Web/sites/metricdefinitions",
                    "Microsoft.Web/sites/metrics",
                    "Microsoft.Web/sites/slots",
                    "Microsoft.Web/sites/slots/instances",
                    "Microsoft.Web/sites/slots/metricdefinitions",
                    "Microsoft.Web/sites/slots/metrics",
                    "Microsoft.Web/staticSites"
                  ],
                  "operator": "In"
                }
              ]
            },
            "policyDefinitionId": "/providers/Microsoft.Authorization/policyDefinitions/a08ec900-254a-4555-9bf5-e42af04b5c5c",
            "policySetDefinitionId": "/providers/Microsoft.Management/managementGroups/learn-sandbox-prod/providers/Microsoft.Authorization/policySetDefinitions/webapps-initiative",
            "policyDefinitionReferenceId": "allowed-resource-types_1",
            "policySetDefinitionName": "webapps-initiative",
            "policyDefinitionName": "a08ec900-254a-4555-9bf5-e42af04b5c5c",
            "policyDefinitionEffect": "deny",
            "policyAssignmentId": "/providers/Microsoft.Management/managementGroups/192fe359-e72a-fa7e-dcd5-95e985e7eac9/providers/Microsoft.Authorization/policyAssignments/webapps-assignment",
            "policyAssignmentName": "webapps-assignment",
            "policyAssignmentScope": "/providers/Microsoft.Management/managementGroups/192fe359-e72a-fa7e-dcd5-95e985e7eac9"
          }
        }
      ]
    }
  ]
}

**Comment by Shreya1414** [comment]
Yes I am facing the same issue. Kindly help.
Regards

**Reply by TheSuranaverse** [reply]
A group for all to get notified for any **Official FRT Announcement, Discuss & Clear any Doubts, Learn & Complete** together.
One-stop place for daily **Events/Workshops + Hackathons + Internships/Jobs/Free Scholarship & Courses + 24/7 Live Music** & much more...

Link to join: https://discord.gg/drQ7ySCdX2
GitHub Solution Forum: https://github.com/github/india/discussions/353

Thank you 🙏

**Comment by ghost** [comment]
Yeah I'm also facing the same problem. I followed the exact steps as shown in unit six, but it is not working for me & I ended by losing today's 2hour of MS Learn Sandbox Activation. Below are some screenshots of problem that i faced.....
![tg](https://user-images.githubusercontent.com/91190563/136653709-076b6e62-4051-4b21-882b-860b6c417f1d.JPG)
![gbhn](https://user-images.githubusercontent.com/91190563/136653731-97b414a9-28ba-403b-b71c-184a248c0b24.JPG)

**Reply by ajeetkbhardwaj** [reply]
> I am facing a problem in creating a website in Unit 6 of the "Describe core Azure architectural components" module. When trying to create the WordPress website as mentioned in the exercise I'm getting the error shown below.
> 
> { "code": "InvalidTemplateDeployment", "message": "The template deployment failed with multiple errors. Please see details for more information.", "details": [ { "code": "RequestDisallowedByPolicy", "target": "ASP-learnac9d3d064a3049e79cd75e7b4b-ac7f", "message": "Resource 'ASP-learnac9d3d064a3049e79cd75e7b4b-ac7f' was disallowed by policy. Policy identifiers: '[{"policyAssignment":{"name":"webapps-assignment","id":"/providers/Microsoft.Management/managementGroups/192fe359-e72a-fa7e-dcd5-95e985e7eac9/providers/Microsoft.Authorization/policyAssignments/webapps-assignment"},"policyDefinition":{"name":"app-service-sku-policy","id":"/providers/Microsoft.Management/managementGroups/learn-sandbox-prod/providers/Microsoft.Authorization/policyDefinitions/app-service-sku-policy"},"policySetDefinition":{"name":"webapps-initiative","id":"/providers/Microsoft.Management/managementGroups/learn-sandbox-prod/providers/Microsoft.Authorization/policySetDefinitions/webapps-initiative"}}]'.", "additionalInfo": [ { "type": "PolicyViolation", "info": { "policyDefinitionDisplayName": "app-service-sku-policy", "policySetDefinitionDisplayName": "webapps-initiative", "evaluationDetails": { "evaluatedExpressions": [ { "result": "True", "expressionKind": "Field", "expression": "type", "path": "type", "expressionValue": "Microsoft.Web/serverfarms", "targetValue": "Microsoft.Web/serverFarms", "operator": "Equals" }, { "result": "True", "expressionKind": "Field", "expression": "Microsoft.Web/serverfarms/sku.name", "path": "sku.name", "expressionValue": "P1v2", "targetValue": [ "F1", "D1", "Y1", "WS1" ], "operator": "NotIn" } ] }, "policyDefinitionId": "/providers/Microsoft.Management/managementGroups/learn-sandbox-prod/providers/Microsoft.Authorization/policyDefinitions/app-service-sku-policy", "policySetDefinitionId": "/providers/Microsoft.Management/managementGroups/learn-sandbox-prod/providers/Microsoft.Authorization/policySetDefinitions/webapps-initiative", "policyDefinitionReferenceId": "app-service-sku-policy_1", "policySetDefinitionName": "webapps-initiative", "policyDefinitionName": "app-service-sku-policy", "policyDefinitionEffect": "Deny", "policyAssignmentId": "/providers/Microsoft.Management/managementGroups/192fe359-e72a-fa7e-dcd5-95e985e7eac9/providers/Microsoft.Authorization/policyAssignments/webapps-assignment", "policyAssignmentName": "webapps-assignment", "policyAssignmentScope": "/providers/Microsoft.Management/managementGroups/192fe359-e72a-fa7e-dcd5-95e985e7eac9" } } ] }, { "code": "RequestDisallowedByPolicy", "target": "CodingClub-mysqldbserver", "message": "Resource 'CodingClub-mysqldbserver' was disallowed by policy. Policy identifiers: '[{"policyAssignment":{"name":"webapps-assignment","id":"/providers/Microsoft.Management/managementGroups/192fe359-e72a-fa7e-dcd5-95e985e7eac9/providers/Microsoft.Authorization/policyAssignments/webapps-assignment"},"policyDefinition":{"name":"Allowed resource types","id":"/providers/Microsoft.Authorization/policyDefinitions/a08ec900-254a-4555-9bf5-e42af04b5c5c"},"policySetDefinition":{"name":"webapps-initiative","id":"/providers/Microsoft.Management/managementGroups/learn-sandbox-prod/providers/Microsoft.Authorization/policySetDefinitions/webapps-initiative"}}]'.", "additionalInfo": [ { "type": "PolicyViolation", "info": { "policyDefinitionDisplayName": "Allowed resource types", "policySetDefinitionDisplayName": "webapps-initiative", "evaluationDetails": { "evaluatedExpressions": [ { "result": "False", "expressionKind": "Field", "expression": "type", "path": "type", "expressionValue": "Microsoft.DBforMySQL/servers", "targetValue": [ "Microsoft.ApiManagement/service", "Microsoft.AlertsManagement/SmartDetectorAlertRules", "Microsoft.Cache/Redis", "Microsoft.Cdn/profiles", "Microsoft.Cdn/profiles/endpoints", "Microsoft.CognitiveServices/accounts", "Microsoft.ContainerRegistry/registries", "Microsoft.ContainerRegistry/registries/webhooks", "Microsoft.DataFactory/factories", "Microsoft.DBforPostgreSQL/servers", "Microsoft.DBforPostgreSQL/serversv2", "Microsoft.DocumentDB/databaseAccounts", "microsoft.insights/actiongroups", "microsoft.insights/components", "Microsoft.KeyVault/vaults", "Microsoft.Logic/workflows", "Microsoft.MixedReality/remoteRenderingAccounts", "Microsoft.Network/connections", "Microsoft.Network/localNetworkGateways", "Microsoft.Network/networkSecurityGroups", "Microsoft.Network/publicIPAddresses", "Microsoft.Network/trafficmanagerprofiles", "Microsoft.Network/virtualNetworkGateways", "Microsoft.Network/virtualNetworks", "Microsoft.Portal/dashboards", "Microsoft.Relay/namespaces", "Microsoft.Resources/resourceGroups", "Microsoft.Search/searchServices", "Microsoft.ServiceBus/namespaces", "Microsoft.ServiceBus/namespaces/authorizationrules", "Microsoft.ServiceBus/namespaces/queues", "Microsoft.ServiceBus/namespaces/queues/authorizationrules", "Microsoft.ServiceBus/namespaces/topics", "Microsoft.ServiceBus/namespaces/topics/authorizationrules", "Microsoft.ServiceBus/namespaces/topics/subscriptions", "Microsoft.ServiceBus/namespaces/topics/subscriptions/rules", "Microsoft.SignalRService/SignalR", "Microsoft.Sql/servers", "Microsoft.Sql/servers/databases", "Microsoft.Storage/storageAccounts", "Microsoft.Web/certificates", "Microsoft.Web/connections", "Microsoft.Web/customApis", "Microsoft.Web/serverFarms", "Microsoft.Web/serverfarms/workers", "Microsoft.Web/sites", "Microsoft.Web/sites/instances", "Microsoft.Web/sites/metricdefinitions", "Microsoft.Web/sites/metrics", "Microsoft.Web/sites/slots", "Microsoft.Web/sites/slots/instances", "Microsoft.Web/sites/slots/metricdefinitions", "Microsoft.Web/sites/slots/metrics", "Microsoft.Web/staticSites" ], "operator": "In" } ] }, "policyDefinitionId": "/providers/Microsoft.Authorization/policyDefinitions/a08ec900-254a-4555-9bf5-e42af04b5c5c", "policySetDefinitionId": "/providers/Microsoft.Management/managementGroups/learn-sandbox-prod/providers/Microsoft.Authorization/policySetDefinitions/webapps-initiative", "policyDefinitionReferenceId": "allowed-resource-types_1", "policySetDefinitionName": "webapps-initiative", "policyDefinitionName": "a08ec900-254a-4555-9bf5-e42af04b5c5c", "policyDefinitionEffect": "deny", "policyAssignmentId": "/providers/Microsoft.Management/managementGroups/192fe359-e72a-fa7e-dcd5-95e985e7eac9/providers/Microsoft.Authorization/policyAssignments/webapps-assignment", "policyAssignmentName": "webapps-assignment", "policyAssignmentScope": "/providers/Microsoft.Management/managementGroups/192fe359-e72a-fa7e-dcd5-95e985e7eac9" } } ] } ] }

That's only the process and you don't have to worry about how to test it practically at this time because. It's the beginning of our learning to fundamentals of Azure IaaS , PaaS and SaaS .there are lots of dependencies to deploy a web apps using the resources and functions and you only deals with SaaS for web apps.

**Reply by TheSuranaverse** [reply]
A group for all to get notified for any **Official FRT Announcement, Discuss & Clear any Doubts, Learn & Complete** together.
One-stop place for daily **Events/Workshops + Hackathons + Internships/Jobs/Free Scholarship & Courses + 24/7 Live Music** & much more...

Link to join: https://discord.gg/drQ7ySCdX2
GitHub Solution Forum: https://github.com/github/india/discussions/353

Thank you 🙏

**Comment by TheSuranaverse** [comment]
A group for all to get notified for any **Official FRT Announcement, Discuss & Clear any Doubts, Learn & Complete** together.
One-stop place for daily **Events/Workshops + Hackathons + Internships/Jobs/Free Scholarship & Courses + 24/7 Live Music** & much more...

Link to join: https://discord.gg/drQ7ySCdX2
GitHub Solution Forum: https://github.com/github/india/discussions/353

Thank you 🙏
